From 677feadf9f6d6d544be27c807d92538297044920 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Tue, 9 Jul 2024 19:02:09 -0300
Subject: [PATCH] nasm: Add -mms-bitfields to the list of ignored flags

Fixes building Nasm objects with Meson's native language support, when
depending against a library that exports that flag, like Glib.

(cherry picked from commit 8b73edd80a3c8ccb8fb6e1f8c8b48fd21175187b)
---
 mesonbuild/compilers/asm.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/compilers/asm.py b/mesonbuild/compilers/asm.py
index d04fbd293..c5e552165 100644
--- a/mesonbuild/compilers/asm.py
+++ b/mesonbuild/compilers/asm.py
@@ -73,7 +73,7 @@ class NasmCompiler(Compiler):
     def unix_args_to_native(self, args: T.List[str]) -> T.List[str]:
         outargs: T.List[str] = []
         for arg in args:
-            if arg == '-pthread':
+            if arg in {'-mms-bitfields', '-pthread'}:
                 continue
             outargs.append(arg)
         return outargs
-- 
2.44.0.windows.1

